Content.Client/UserInterface/GhostGui.cs

```diff
@@ -18,18 +25,133 @@ public GhostGui(GhostComponent owner)
 
             _owner = owner;
 
+            var targetMenu = new GhostTargetWindow(_playerManager,owner);
+
             MouseFilter = MouseFilterMode.Ignore;
 
+            GhostWarp.OnPressed += (args) => { targetMenu.Populate(); };
+
             ReturnToBody.OnPressed += (args) => { owner.SendReturnToBodyMessage(); };
 
-            AddChild(ReturnToBody);
+
+            AddChild((new HBoxContainer
+            {
+                Children =
+                {
+                    ReturnToBody,
+                    GhostWarp
+                }
+            }));
 
             Update();
         }
 
+
         public void Update()
         {
             ReturnToBody.Disabled = !_owner.CanReturnToBody;
         }
     }
+
+    internal class GhostTargetWindow : SS14Window
+    {
+
+        protected override Vector2? CustomSize => (300, 450);
+        private readonly IPlayerManager _playerManager;
+        private List<Button> _buttons = new List<Button>();
+        private readonly GhostComponent _owner;
+        private VBoxContainer _buttonContainer;
+
+        public GhostTargetWindow(IPlayerManager manager,GhostComponent owner)
+        {
+            Title = "Ghost Warp";
+            _playerManager = manager;
+            _owner = owner;
+            _owner.GhostRequestWarpPoint();
+            _owner.GhostRequestPlayerNames();
+
+            var margin = new MarginContainer()
+            {
+                SizeFlagsVertical = SizeFlags.FillExpand,
+                SizeFlagsHorizontal = SizeFlags.FillExpand,
+            };
+
+            _buttonContainer = new VBoxContainer()
+            {
+                SizeFlagsVertical = SizeFlags.FillExpand,
+                SizeFlagsHorizontal = SizeFlags.Fill,
+                SeparationOverride = 5,
+
+            };
+
+            var _scrollBarcontainer = new ScrollContainer()
```

Comment:
That is against the Geneva C# convention
```suggestion
            var scrollBarcontainer = new ScrollContainer()
```